Frameworks/ImageIO/CGImageSource.mm

```diff
@@ -489,21 +1370,109 @@ size_t CGImageSourceGetCount(CGImageSourceRef isrc) {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes A limited subset of source container properties are supported currently  
 */
 CFDictionaryRef CGImageSourceCopyProperties(CGImageSourceRef isrc, CFDictionaryRef options) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    RETURN_NULL_IF(!isrc);
+    NSData* imageData = ((ImageSource*)isrc).data;
+    RETURN_NULL_IF(!imageData);
+
+    NSMutableDictionary *properties = [[NSMutableDictionary alloc] init]; 
+    [properties setObject:[NSNumber numberWithInt:[imageData length]] forKey:(id)kCGImagePropertyFileSize];
+
+    // Look for container-level metadata if the file is a GIF.
```

Comment:
From the MSDN Documentation on the metadata query reader: 

If an image format does not support container-level metadata, this will return WINCODEC_ERR_UNSUPPORTEDOPERATION. The only Windows provided image format that supports container-level metadata is GIF. Instead, use IWICBitmapFrameDecode::GetMetadataQueryReader.

Only GIF can have metadata for the whole entire image source, which is what this API looks for. Individual frame properties are handled by the CGImageSourceCopyPropertiesAtIndex API.